src/shapes/Object/ObjectGeometry.ts

```diff
@@ -437,40 +438,29 @@ export class ObjectGeometry<EventSpec extends ObjectEvents = ObjectEvents>
     this.aCoords = this.calcACoords();
   }
 
-  transformMatrixKey(skipGroup = false): string {
-    const sep = '_';
-    let prefix = '';
+  transformMatrixKey(skipGroup = false): number[] {
+    let prefix: number[] = [];
     if (!skipGroup && this.group) {
-      prefix = this.group.transformMatrixKey(skipGroup) + sep;
+      prefix = this.group.transformMatrixKey(skipGroup);
     }
-    return (
-      prefix +
-      this.top +
-      sep +
-      this.left +
-      sep +
-      this.scaleX +
-      sep +
-      this.scaleY +
-      sep +
-      this.skewX +
-      sep +
-      this.skewY +
-      sep +
-      this.angle +
-      sep +
-      this.originX +
-      sep +
-      this.originY +
-      sep +
-      this.width +
-      sep +
-      this.height +
-      sep +
-      this.strokeWidth +
-      this.flipX +
-      this.flipY
+    prefix.push(
+      this.top,
+      this.left,
+      this.scaleX,
+      this.scaleY,
+      this.skewX,
+      this.skewY,
+      this.angle,
+      resolveOrigin(this.originX),
+      resolveOrigin(this.originY),
+      this.width,
+      this.height,
+      +this.flipX,
+      +this.flipY,
+      this.strokeWidth
```

Comment:
i would argue that angle could go right away top and left, then width, height, scaleX and scaleY and then all the rest. Origin last. Is probably non influent but i would have the properties that have a chance to change more often at the top of the array.

I also think this list could be flawed for the exactbounding box case

Reply:
Done, I've sorted properties by likelihook of change. I put angle after scale as I think it's less probable. People usually move, resize, scale objects and then less frequently rotate them.

An advantage of this method is also that it's easy to override in the app and sort/remove the properties according to the app.

> I also think this list could be flawed for the exactbounding box case

Not sure to understand what you're referring and what would be the action